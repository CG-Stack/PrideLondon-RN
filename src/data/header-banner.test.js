// @flow
import decodeHeaderBanner from "./header-banner";

describe("HeaderBanner", () => {
  describe("decoder", () => {
    it("correctly decodes valid CMS headerbanner", () => {
      const data: mixed = {
        fields: {
          heading: { "en-GB": "heading" },
          headingLine2: { "en-GB": "headingLine2" },
          subHeading: { "en-GB": "subHeading" },
          heroImage: { "en-GB": { sys: { id: "2o2SZPgYl2ABCWu2MoK333" } } },
          backgroundColour: { "en-GB": "#333333" }
        },
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "headerBanner"
            }
          },
          revision: 1
        }
      };

      const decoded = decodeHeaderBanner("en-GB")(data);
      expect(decoded.ok).toEqual(true);
      if (decoded.ok) {
        expect(decoded.value).toMatchSnapshot();
      }
    });

    it("fails if a property is missing", () => {
      const data: mixed = {
        fields: {},
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "headerBanner"
            }
          }
        }
      };

      const decoded = decodeHeaderBanner("en-GB")(data);
      expect(decoded.ok).toEqual(false);
    });
  });
});
