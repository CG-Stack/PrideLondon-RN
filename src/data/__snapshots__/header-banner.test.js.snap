// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBanner decoder correctly decodes valid CMS headerbanner 1`] = `
Object {
  "contentType": "headerBanner",
  "fields": Object {
    "backgroundColour": "#333333",
    "heading": "heading",
    "headingLine2": "headingLine2",
    "heroImage": Object {
      "sys": Object {
        "id": "2o2SZPgYl2ABCWu2MoK333",
      },
    },
    "subHeading": "subHeading",
  },
  "id": "3O3SZPgYl2MUEWu2MoK2oi",
  "locale": "en-GB",
  "revision": 1,
}
`;
